crates/parser/test_data/parser/ok/0073_associated_const_equality.rs

```diff
@@ -0,0 +1,3 @@
+impl<T, const N: usize> Foo for T where T: HasCount<COUNT = {N}> {}
+
+impl<T> Foo for T where T: HasCount<COUNT = 0> {}
```

Comment:
These tests should be written as comments in the source code and then auto-generated by running `cargo test -p syntax`. See the `// test test_name` comments in the parser modules.

Reply:
I have written these tests as comments and changed them a bit.  
It seems that `cargo test -p syntax` doesn't automatically generate tests files, instead, I manually run `sourcegen_parser_tests`

Reply:
Ah my bad, its `cargo test -p parser`